src/runtime/auto-prompt-manager.ts

```diff
@@ -271,22 +280,28 @@ export class AutoPromptManager {
 
   /**
    * Determines whether a mini prompt for contributions or subscriptions should
-   * be shown.
+   * be shown based on the UI Predicates.
    */
-  async shouldShowAutoPrompt_(
-    clientConfig: ClientConfig,
-    entitlements: Entitlements,
-    autoPromptType?: AutoPromptType
-  ): Promise<boolean> {
+  shouldShowMonetizationPromptFromUiPredicates(
+    uiPredicates?: UiPredicates
+  ): boolean {
     // If false publication predicate was returned in the response, don't show
     // the prompt.
-    if (
-      clientConfig.uiPredicates &&
-      !clientConfig.uiPredicates.canDisplayAutoPrompt
-    ) {
-      return Promise.resolve(false);
+    if (uiPredicates && !uiPredicates.canDisplayAutoPrompt) {
+      return false;
     }
 
+    return true;
+  }
+
+  /**
+   * Determines whether a mini prompt for contributions or subscriptions should
+   * be shown based on the frequency cap.
+   */
+  async shouldShowMonetizationPromptFromFrequencyCap(
```

Comment:
ditto. what about `canDisplayMonetizationPromptAfterFrequencyCapping`?

Reply:
updated to `canDisplayMonetizationPromptAfterFrequencyCap_`